### PR TITLE
demo/billing: port to mzworkflows.py

### PR DESCRIPTION
**Pull request by benesch**
Touches #9809. Since the demo is no longer used as a load test for
releases, the load test workflows are not ported over.

### Motivation

   * This PR refactors existing code.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR adds a release note for any [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/user/content/release-notes.md#what-changes-require-a-release-note).

### CODE REVIEW

File: doc/user/content/demos/microservice.md [hunk 1]
```diff
@@ -436,7 +436,7 @@ can see that Materialize is ingesting the `protobuf` data and normalizing it.
 1. Launch `psql` (or any Materialize CLI client), connecting to `materialize`:
 
     ```shell
-    psql -U materialize -h localhost -p 6875 -d materialize
+    psql -U materialize -h localhost -p $(./mzcompose list-ports materialized) -d materialize
```
Comment (ruchirK) [inline doc/user/content/demos/microservice.md:439]: do we still need to specify `-d materialize`? (not your change i know)

File: demo/billing/mzcompose.yml [hunk 2]
```diff
@@ -7,167 +7,48 @@
...
-    ports:
-     - *materialized
-    command: --workers ${MZ_WORKERS:-1} --differential-idle-merge-effort=1000 --disable-telemetry
+    - db-data:/share/billing-demo/data
+    ports: [6875]
+    command: --disable-telemetry
     environment:
-      - MZ_LOG_FILTER=dataflow=error,info
-      - MZ_DEV=1
+    - MZ_LOG_FILTER=dataflow=error,info
```
Comment (ruchirK) [inline demo/billing/mzcompose.yml:20]: the error log level for dataflow seems odd, and i feel like we could just get rid of it?

**Review by ruchirK — APPROVED** [review]
LGTM!

File: demo/billing/mzcompose.yml (hunk 2 above)
Comment (benesch) [reply demo/billing/mzcompose.yml:20]: Done.

File: doc/user/content/demos/microservice.md (hunk 1 above)
Comment (benesch) [reply doc/user/content/demos/microservice.md]: We do! I added a `./mzcompose sql` command to make this *much* easier though.

**Pull request merged**